tour_manager/src/main/java/edu/artic/tours/manager/TourProgressManager.kt

```diff
@@ -23,4 +27,57 @@ class TourProgressManager {
     val selectedTour: Subject<Optional<ArticTour>> = BehaviorSubject.createDefault(Optional(null))
     val proposedTour: Subject<Optional<Pair<ArticTour, ArticTour.TourStop>>> = BehaviorSubject.createDefault(Optional(null))
     val leaveTourRequest: Subject<Boolean> = PublishSubject.create()
+
+    /**
+     * If [audioFileModel] belongs to currently selected tour i.e. [selectedTour],
+     * this method advances [selectedStop] to next tour stop.
+     */
+    @SuppressLint("CheckResult")
+    fun playBackEnded(audioFileModel: AudioFileModel) {
+        Observables.combineLatest(selectedTour.filterValue(), selectedStop)
+                .take(1)
+                .subscribe { (tour, currentStopID) ->
+
+                    /**
+                     * Get audio file id
+                     */
+                    val audioFileID: String? = if (currentStopID == INTRO_TOUR_STOP_OBJECT_ID) {
+                        tour.tourAudio
+                    } else {
+                        tour.tourStops.find { it.objectId == currentStopID }?.audioId
+                    }
+
+                    audioFileID?.let { audioID ->
+                        audioFileDao
+                                .getAudioByIdAsync(audioID)
```

Comment:
What do you think of making `getAudioByIdAsync` return `Observable` instead of `Flowable`? That way we won't need the `::toObservable` calls here and in `TourCarousalIntroViewModel`.

Reply:
Room database does not support `Observable`. Please see https://developer.android.com/training/data-storage/room/accessing-data#query-observable
 
Leaving this as it is.

Reply:
Ah, that support was added to Room 2.0.0 back in july (see note in https://medium.com/androiddevelopers/room-rxjava-acb0cd4f375). We can take advantage of it whenever we upgrade to that or a higher version.